### DISCUSSION TITLE
Is Client-Side Recording, Server-Side Transcription Setup Possible?

### DISCUSSION THREAD

**Discussion by Hu-nie** [KoljaB/RealtimeSTT#148] [Q&A]
The current structure involves the client sending only a flag to initiate recording, while the server handles both recording and transcription.

Is it possible to change this setup so that the client handles recording using PyAudio, sends the audio data over a socket to the server, and then the server only processes the transcription?

**Comment by KoljaB** [comment]
Yes, that's possible. Please look into the RealtimeSTT_server folder. There's a Readme in there with details. 
The stt-server can consume audiochunks and both the stt cli command as well as the AudioToTextRecorderClient class in the RealtimeSTT folder would record with pyaudio and send the chunks from the client to that stt-server.

**Reply by Hu-nie** [reply]
Oh, thanks! @KoljaB 


But it seems that having multiple users connected at the same time is still difficult, right?

**Reply by KoljaB** [reply]
Yes, that's correct.

**Reply by Hu-nie** [reply]
I’m trying to develop multi-user connections, but can you tell me why it might not be working?
@KoljaB

**Reply by KoljaB** [reply]
1. faster_whisper doesn't allow concurrent transcription on a single GPU (it can do that for CPU only and on multiple GPU setups but that's not enough for RealtimeSTT) 
2. Realtime transcription takes lots of resources. We'd need to load balance that. 

These arent really that hard to solve problems. I didn't have the need personally to make it multi-user ready yet. 

I currently offer (paid) consulting services to companies interested in implementing advanced real-time transcription and TTS solutions, including load balancing, multi-user support, and optimizing AI model deployment for resource efficiency.

**Reply by sangheonEN** [reply]
If I set num_workers: int = number of users in the WhisperModel object of Faster whisper, is it impossible to support multiple users?

class WhisperModel:
def __init__(
self,
model_size_or_path: str,
device: str = "auto",
device_index: Union[int, List[int]] = 0,
compute_type: str = "default",
cpu_threads: int = 0,
num_workers: int = 1,
download_root: Optional[str] = None,
local_files_only: bool = False,
files: dict = None,
**model_kwargs,
):
